### PR TITLE
CeresScanMatcher target_translation

### PR DESCRIPTION
**Pull request by gaschler**
For its translation cost function, CeresScanMatcher now takes
a target translation as an argument instead of a pose, which 
was confusing.

### CODE REVIEW

File: cartographer/mapping_3d/local_trajectory_builder.cc [hunk 1]
```diff
@@ -158,7 +158,7 @@ LocalTrajectoryBuilder::AddAccumulatedRangeData(
     return nullptr;
   }
   ceres_scan_matcher_->Match(
-      matching_submap->local_pose().inverse() * pose_prediction,
+      (matching_submap->local_pose().inverse() * pose_prediction).translation(),
```
Comment (cschuet) [inline cartographer/mapping_3d/local_trajectory_builder.cc]: Can we write this more efficiently now?

Comment (gaschler) [reply cartographer/mapping_3d/local_trajectory_builder.cc]: Done

Comment (gaschler) [reply cartographer/mapping_3d/local_trajectory_builder.cc]: Actually, I reverted the more efficient computation because it was less readable.

**Comment by gaschler** [comment]
@wally-the-cartographer merge

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user gaschler. Merge queue now has a length of 1.

**Pull request merged**